x-pack/plugins/triggers_actions_ui/public/application/sections/rule_form/rule_form_advanced_options.tsx

```diff
@@ -0,0 +1,316 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useCallback, useMemo } from 'react';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n-react';
+import {
+  EuiBadge,
+  EuiFieldNumber,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiFormRow,
+  EuiIconTip,
+  EuiPanel,
+  EuiSwitch,
+  EuiText,
+  useIsWithinMinBreakpoint,
+  useEuiTheme,
+  EuiHorizontalRule,
+  EuiSpacer,
+  EuiSplitPanel,
+  EuiLoadingSpinner,
+} from '@elastic/eui';
+import { RuleSettingsFlappingInputs } from '@kbn/alerts-ui-shared/src/rule_settings/rule_settings_flapping_inputs';
+import {
+  RuleSettingsFlappingMessage,
+  flappingOffMessage,
+} from '@kbn/alerts-ui-shared/src/rule_settings/rule_settings_flapping_message';
+import { RuleSpecificFlappingProperties } from '@kbn/alerting-plugin/common';
+import { useGetFlappingSettings } from '../../hooks/use_get_flapping_settings';
+
+const flappingLabel = i18n.translate(
+  'xpack.triggersActionsUI.ruleFormAdvancedOptions.flappingLabel',
+  {
+    defaultMessage: 'Flapping Detection',
+  }
+);
+
+const flappingOnLabel = i18n.translate('xpack.triggersActionsUI.ruleFormAdvancedOptions.onLabel', {
+  defaultMessage: 'ON',
+});
+
+const flappingOffLabel = i18n.translate(
+  'xpack.triggersActionsUI.ruleFormAdvancedOptions.offLabel',
+  {
+    defaultMessage: 'OFF',
+  }
+);
+
+const flappingOverrideLabel = i18n.translate(
+  'xpack.triggersActionsUI.ruleFormAdvancedOptions.overrideLabel',
+  {
+    defaultMessage: 'Override',
+  }
+);
+
+const flappingOverrideConfiguration = i18n.translate(
+  'xpack.triggersActionsUI.ruleFormAdvancedOptions.flappingOverrideConfiguration',
+  {
+    defaultMessage: 'Override Configuration',
+  }
+);
+
+const clampFlappingValues = (flapping: RuleSpecificFlappingProperties) => {
+  return {
+    ...flapping,
+    statusChangeThreshold: Math.min(flapping.lookBackWindow, flapping.statusChangeThreshold),
+  };
+};
+
+const INTEGER_REGEX = /^[1-9][0-9]*$/;
+
+export interface RuleFormAdvancedOptionsProps {
+  alertDelay?: number;
+  flappingSettings?: RuleSpecificFlappingProperties;
+  onAlertDelayChange: (value: string) => void;
+  onFlappingChange: (value: RuleSpecificFlappingProperties | null) => void;
+  enabledFlapping?: boolean;
+}
+
+export const RuleFormAdvancedOptions = (props: RuleFormAdvancedOptionsProps) => {
+  const {
+    alertDelay,
+    flappingSettings,
+    enabledFlapping = false,
```

Comment:
I'm thinking about the serverless rollback restrictions.

How about making `enabledFlapping` something like `!isEmpty(flappingSettings)` for an initial release?

I see that `flappingSettings={rule.flapping}` so if someone created a rule with flapping settings they could still edit it in case of a rollback.

We could do the following:
Serverless release 1 - Merge the frontend
Serverless release 2 - Merge the backend

In case of a rollback from 2 to 1 rules with flapping could still be edited in the UI.

Wdyt? (maybe @cnasikas too)

Reply:
Good point @adcoelho! Based on the immediate release process the backend changes should be always in the first serverless release. The reason is if the task manager picks up a task on release 2 (with the new flapping property) and we rollback to 1 it will start failing because the flapping property would not be there. I would start with the backend. It will also help us understand better how the schema would look like and what restrictions it may impose in the UI. We can create a feature branch, merge the backend PR on there, then merge the UI, and finally implement the immediate release process.

Reply:
So we would have:
1. backend release
2. UI release where the form is only visible if the rule has `flappingSettings` set
3. make form is **always** visible

?

Reply:
Yeap!

Reply:
Also flapping is going to be a nullable property so it should work without it being defined (rule specific flapping will be created lazily)